pkg/console/controllers/oidcsetup/oidcsetup.go

```diff
@@ -0,0 +1,280 @@
+		return err
+	} else if !shouldSync {
+		return nil
+	}
+
+	oidcClientsSchema, err := authnConfigHasOIDCFields(c.crdLister)
+	if err != nil {
+		return statusHandler.FlushAndReturn(err)
+	}
+
+	// the schema is feature-gating this controller, we assume API validation won't
+	// allow authentication/cluster 'Type=OIDC' if the `.status.oidcClients` field
+	// does not exist
+	if !oidcClientsSchema {
+		// reset all conditions set by this controller
+		statusHandler.AddConditions(status.HandleProgressingOrDegraded("OIDCClientConfig", "", nil))
+		statusHandler.AddConditions(status.HandleProgressingOrDegraded("AuthStatusHandler", "", nil))
+		return statusHandler.FlushAndReturn(nil)
+	}
+
+	operatorConfig, err := c.consoleOperatorLister.Get(api.ConfigResourceName)
+	if err != nil {
+		return err
+	}
+
+	authnConfig, err := c.authnLister.Get(api.ConfigResourceName)
+	if err != nil {
+		return err
+	}
+
+	// we need to keep track of errors during the sync so that we can requeue
+	// if any occur
+	var errs []error
+	if authnConfig.Spec.Type == configv1.AuthenticationTypeOIDC {
+		syncErr := c.syncAuthTypeOIDC(ctx, syncCtx, statusHandler, operatorConfig, authnConfig)
+		statusHandler.AddConditions(
+			status.HandleProgressingOrDegraded(
+				"OIDCClientConfig", "OIDCConfigSyncFailed",
+				syncErr,
+			),
+		)
+		if syncErr != nil {
+			errs = append(errs, syncErr)
+		}
+	}
+
+	applyErr := c.authStatusHandler.Apply(ctx, authnConfig)
+	statusHandler.AddConditions(status.HandleProgressingOrDegraded("AuthStatusHandler", "FailedApply", applyErr))
+	if applyErr != nil {
+		errs = append(errs, applyErr)
+	}
+
+	if len(errs) > 0 {
+		return statusHandler.FlushAndReturn(factory.SyntheticRequeueError)
+	}
+	return statusHandler.FlushAndReturn(nil)
+}
+
+func (c *oidcSetupController) syncAuthTypeOIDC(
+	ctx context.Context,
+	controllerContext factory.SyncContext,
+	statusHandler status.StatusHandler,
+	operatorConfig *operatorv1.Console,
+	authnConfig *configv1.Authentication,
+) error {
+
+	clientConfig := utilsub.GetOIDCClientConfig(authnConfig)
+	if clientConfig == nil {
+		c.authStatusHandler.WithCurrentOIDCClient("")
+		c.authStatusHandler.Unavailable("OIDCClientConfig", "no OIDC client found")
+		return nil
+	}
+
+	if len(clientConfig.ClientID) == 0 {
+		return fmt.Errorf("no ID set on console's OIDC client")
+	}
+	c.authStatusHandler.WithCurrentOIDCClient(clientConfig.ClientID)
+
+	if len(clientConfig.ClientSecret.Name) == 0 {
+		c.authStatusHandler.Degraded("OIDCClientMissingSecret", "no client secret in the OIDC client config")
+		return nil
+	}
+
+	clientSecret, err := c.targetNSSecretsLister.Secrets(api.TargetNamespace).Get("console-oauth-config")
+	if err != nil {
+		c.authStatusHandler.Degraded("OIDCClientSecretGet", err.Error())
+		return err
+	}
+
+	if valid, msg, err := c.checkClientConfigStatus(authnConfig, clientSecret); err != nil {
+		c.authStatusHandler.Degraded("DeploymentOIDCConfig", err.Error())
+		return err
```

Comment:
I haven't yet looked, but I bet that this will cause the progressing and available conditions on the authentication.status.oidcclients[console] to disappear.

Reply:
https://github.com/openshift/console-operator/blob/68e37a85d0d3a083c396f7654707b49b72553c49/pkg/console/status/auth_status.go#L119-L127 should be handling that I beleive

Reply:
If we take this early return, what sets the Progressing and Available condition since the authStatusHandler self-clears the conditions here: on every apply.

I typed all the way to here before realizing that the combination of stateful, plus early return on error protected us.

this really needs to be transitioned to per-invocation scoped, not instance scoped for ease of reading+maintenance.  I'm pretty good at this and I can't read it.